wkcuber/api/MagDataset.py

```diff
@@ -0,0 +1,135 @@
+from os.path import join
+
+from wkw import wkw
+
+import wkcuber.api as api
+from wkcuber.api.Slice import WKSlice, TiffSlice
+from wkcuber.api.TiffData.TiffMag import TiffMagHeader
+
+
+class MagDataset:
+    def __init__(self, layer, name):
+        self.mag = {}
+        self.layer = layer
+        self.name = name
+        self.header = self.get_header()
+
+        file_path = join(self.layer.dataset.path, self.layer.name, self.name)
+        size = self.layer.dataset.properties.data_layers[
```

Comment:
What's the difference between `self.layer` and `self.layer.dataset.properties.data_layers[self.layer.name]`? Adding python type annotations might be helpful here, since the types also act as documentation.